### PR TITLE
fix(combobox): selecting from filtered list Al 5164

### PR DESCRIPTION
**Pull request by ianaa**
fixes a bug introduced in [v2.6.4](https://github.com/archilogic-com/honeycomb/releases/tag/v2.6.4). Looks like the query was cleared before the model has been updated.
Added a couple of story interaction tests for this path.
This will trigger a storybook 7 update to be release and I hope will fix the issue with the docs pages.

### CODE REVIEW

File: src/components/Combobox.vue [hunk 1]
```diff
@@ -128,13 +128,21 @@ export default defineComponent({
       }
     }
 
+    let blurTimeout: NodeJS.Timeout
```
Comment (manuatarchilogic) [inline src/components/Combobox.vue]: you sure about this type?

File: src/components/Combobox.vue [hunk 2]
```diff
@@ -128,13 +128,21 @@ export default defineComponent({
...
+    const clearQuery = () => {
+      blurTimeout = setTimeout(() => {
+        query.value = ''
+        emit('update:query', query.value)
+      }, 100)
```
Comment (manuatarchilogic) [inline src/components/Combobox.vue]: wouldn't this be a typical case for `await nextTick()`?

**Review by manuatarchilogic — APPROVED** [review]

File: src/components/Combobox.vue (hunk 2 above)
Comment (ianaa) [reply src/components/Combobox.vue]: it has to be more than just next tick or setTimeout with 0 delay. Maybe the model gets updated on the next tick and the onblur handler still gets called first?...

File: src/components/Combobox.vue (hunk 1 above)
Comment (ianaa) [reply src/components/Combobox.vue]: it's one of possible options (another is type `number` if using `window.setTimeout`).
but actually it looks like I might not need it at all, clearing the timeout might be superfluous - seems that delaying the timeout is the key here.

File: src/components/Combobox.vue (hunk 2 above)
Comment (manuatarchilogic) [reply src/components/Combobox.vue]: yeah, I dunno. Just feels weird relying on a fixed delay

**Review by romiekos — APPROVED** [review]

File: src/components/Combobox.vue (hunk 2 above)
Comment (ianaa) [reply src/components/Combobox.vue]: well I'm not gonna try calling it in a double `nextTick` :D

**Pull request merged**